app/helpers/application_helper.rb

```diff
@@ -39,25 +39,40 @@ def passwd_url
     ENV['OOD_DASHBOARD_PASSWD_URL']
   end
 
-  def fa_icon(icon)
-    content_tag(:i, "", class: ["fa", "fa-#{icon}", "fa-fw", "app-icon"] , title: "FontAwesome icon specified: #{icon}")
+  def fa_icon(icon, fa_style = "fas")
+    content_tag(:i, "", class: [fa_style, "fa-#{icon}", "fa-fw", "app-icon"] , title: "FontAwesome icon specified: #{icon}")
   end
 
   def app_icon_tag(app)
     if app.icon_path.file?
       image_tag app_icon_path(app.name, app.type, app.owner), class: 'app-icon', title: app.icon_path
     else # default to font awesome icon
-      icon = (app.manifest.icon =~ /fa:\/\/(.*)/) ? $1 : "gear"
-      fa_icon(icon)
+      if app.manifest.icon =~ /^fa[bsrl]?:\/\/(.*)/
```

Comment:
you can wrap both matches in one regex:

```ruby
/^(fa[bsrl]?):\/\/(.*)/

$1 # == "fa"
$2 # == "icon"
```

Also this is interesting. I'd prefer to make the default case easiest. For example, in the future, if I did `fa://cog` I would like to see that translate to the class `fas fa-cog`. Because in the default case I just want to choose an icon, and not think about the icon style. In fact, we could stay we do not support different icon styles, only the "solid style".

But I gather that would actually be difficult because then you would need some logic to determine which icons are safe to use the old style? Does the shim make every single icon work with the class format "fas fa-ICON" such as "fas fa-gear"?

Reply:
Or, if `fa://` is used then we default to `fas` for the style... is that possible? I guess it depends on how the shims work.

Reply:
`fas fa-gear` does not work; it must be `fa fa-gear`. That is why I think we should move on to changing the way the URI works and leave `fa://` only for backward compatibility.

Reply:
Oh but according to https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4 `fa fa-cog` should work i.e. `Font Awesome Solid == fas or fa`. Furthermore, `far` and `fal` will only work if we have the pro version, which we do not (cost $$).`fab` would work though.

Can you confirm that all the icons you changed to `fas` would work if just using `fa`?

Reply:
Actually, there are some free icons under `far`, but none for `fal`. But it does work if you use the new name for the icon and the old `fa` class. So default everything to just `fa`?